### ISSUE TITLE
Panic induced by use of `unAnnotateS`

### ISSUE THREAD

**Issue by Gabriel439** [quchen/prettyprinter#50]
Here is a small reproducing example:

```
$ nix-shell --packages 'haskellPackages.ghcWithPackages (pkgs: [ pkgs.prettyprinter ])' --run ghci
GHCi, version 8.0.2: http://www.haskell.org/ghc/  :? for help
>>> :set -XOverloadedStrings
>>> import Data.Text.Prettyprint.Doc
>>> import Data.Text.Prettyprint.Doc.Render.Text
>>> renderStrict (unAnnotateS (layoutSmart defaultLayoutOptions (annotate True "a")))
"*** Exception: An unpaired style terminator was encountered. This is a bug in the layout algorithm! Please report this as a bug
CallStack (from HasCallStack):
  error, called at src/Data/Text/Prettyprint/Doc/Render/Util/Panic.hs:18:20 in prettyprinter-1.1.1-1CDqnG9d6HQ5GZzz2F5LpU:Data.Text.Prettyprint.Doc.Render.Util.Panic
```

**Comment by quchen** [comment]
Uuuh yeah `unAnnotateS` is horribly broken, because it’s defined in terms of `alterAnnotationsS`, which is, you guessed it, horribly broken.

**Comment by quchen** [comment]
Fixed, thanks!

**Comment by Gabriel439** [comment]
You're welcome! :)

**Comment by quchen** [comment]
…released 1.2.0.1 of Prettyprinter
